src/contracts/deposits-registry/fetcher/fetcher.service.ts

```diff
@@ -0,0 +1,124 @@
+import { Injectable } from '@nestjs/common';
+import { BlsService } from 'bls';
+import { RepositoryService } from 'contracts/repository';
+import { DepositEventEvent } from 'generated/DepositAbi';
+
+import { ProviderService } from 'provider';
+import { parseLittleEndian64 } from '../crypto';
+import { DEPLOYMENT_BLOCK_NETWORK } from '../deposits-registry.constants';
+import { DepositEvent, VerifiedDepositEventGroup } from '../interfaces';
+import { DepositTree } from '../sanity-checker/integrity-checker/deposit-tree';
+
+@Injectable()
+export class DepositsRegistryFetcherService {
+  constructor(
+    private providerService: ProviderService,
+    private repositoryService: RepositoryService,
+    private blsService: BlsService,
+  ) {}
+
+  /**
+   * Returns events in the block range
+   * If the request failed, it tries to repeat it or split it into two
+   * @param startBlock - start of the range
+   * @param endBlock - end of the range
+   * @returns event group
+   */
+  public async fetchEventsFallOver(
+    startBlock: number,
+    endBlock: number,
+  ): Promise<VerifiedDepositEventGroup> {
+    return await this.providerService.fetchEventsFallOver(
+      startBlock,
+      endBlock,
+      this.fetchEvents.bind(this),
+    );
+  }
+
+  /**
+   * Returns events in the block range
```

Comment:
It also verifies deposits, so it's worth adding. maybe even in the name of method